src/lazy/encoder/binary/v1_1/fixed_uint.rs

```diff
@@ -65,6 +65,20 @@ impl FixedUInt {
     pub fn size_in_bytes(&self) -> usize {
         self.size_in_bytes
     }
+
+    /// Write the provided UInt-like as a uint8 ensuring that the value fits in 8bits.
+    #[inline]
+    pub(crate) fn write_as_uint8<W: Write>(output: &mut W, value: impl Into<UInt>) -> IonResult<()> {
```

Comment:
Why can't we use `value: impl Into<u8>`? I'm sure you have a good reason, but I don't know what that is and my gut reaction is that it seems like accepting `impl Into<UInt>` is just concealing a potential problem from the users.

Reply:
> I could switch this over to use a TryInto<u8>, and use the implementation to validate the range of the underlying value. 

No, don't worry about it. I see also that this is a `pub(crate)` function, so even if we were to change it, it would need to be changed elsewhere (public signatures) in order for the user to actually get any benefit.